### PR TITLE
feat: `ListWalletTransactions` rpc

### PR DESCRIPTION
**Pull request by jackstar12**

### CODE REVIEW

File: boltzrpc/boltzrpc.proto [hunk 1]
```diff
@@ -733,6 +738,39 @@ message GetWalletRequest {
     optional uint64 id = 2;
 }
 
+message ListWalletTransactionsRequest {
+    uint64 id = 1;
+    optional bool exclude_swap_related = 2;
+}
+
+enum TransactionType {
+    ANY = 0;
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:747]: What does `ANY` mean?

File: boltzrpc/boltzrpc.proto [hunk 2]
```diff
@@ -733,6 +738,39 @@ message GetWalletRequest {
...
+    LOCKUP = 1;
+    REFUND = 2;
+    CLAIM = 3;
+    RECEIVE = 4;
+    SEND = 5;
+}
+
+message WalletTransaction {
+    string tx_id = 1;
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:756]:
```suggestion
    string id = 1;
```

File: boltzrpc/boltzrpc.proto [hunk 3]
```diff
@@ -733,6 +738,39 @@ message GetWalletRequest {
...
+    uint64 total_amount = 2;
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:757]: Is that the sum of the inputs or outputs?

File: boltzrpc/boltzrpc.proto [hunk 4]
```diff
@@ -733,6 +738,39 @@ message GetWalletRequest {
...
+    int64 timestamp = 3;
+    repeated TransactionOutput outputs = 4;
+    uint32 block_height = 5;
+    optional string swap_id = 6;
+    TransactionType type = 7;
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:762]: What if we ever do batching; multiple swap related actions in a single transaction?

File: database/any.go [hunk 5]
```diff
@@ -104,3 +112,20 @@ func (database *Database) QueryAllSwaps(args SwapQuery) ([]*AnySwap, error) {
...
+	for i, _ := range transactionIds {
+		placeholders = append(placeholders, fmt.Sprintf("$%d", i+1))
+	}
+	placeholder := strings.Join(placeholders, ", ")
+	values := make([]any, len(transactionIds))
+	for i, txId := range transactionIds {
+		values[i] = txId
+	}
+	where, values := args.ToWhereClauseWithExisting(
+		[]string{fmt.Sprintf("lockupTransactionid IN (%s) OR claimTransactionid IN (%s) OR refundTransactionid IN(%s)", placeholder, placeholder, placeholder)},
```
Comment (michael1011) [inline database/any.go:127]:
```suggestion
		[]string{fmt.Sprintf("lockupTransactionid IN (%s) OR claimTransactionid IN (%s) OR refundTransactionid IN (%s)", placeholder, placeholder, placeholder)},
```

**Review by michael1011 — APPROVED** [review]

**Review by michael1011 — COMMENTED** [review]
Misclicked the approve

File: boltzrpc/boltzrpc.proto (hunk 4 above)
Comment (jackstar12) [reply boltzrpc/boltzrpc.proto:762]: we can make it repeated. the type can stay the same unless we start to mix claims and refunds together

File: boltzrpc/boltzrpc.proto (hunk 3 above)
Comment (jackstar12) [reply boltzrpc/boltzrpc.proto:757]: Outputs

File: boltzrpc/boltzrpc.proto (hunk 1 above)
Comment (jackstar12) [reply boltzrpc/boltzrpc.proto:747]: Thought it could ne useful if the tx type cant be identified clearly

File: boltzrpc/boltzrpc.proto (hunk 3 above)
Comment (michael1011) [reply boltzrpc/boltzrpc.proto:757]: With out without change output?

Comment (jackstar12) [reply boltzrpc/boltzrpc.proto:757]: without change.

Comment (jackstar12) [reply boltzrpc/boltzrpc.proto:757]: what about a `net_amount` which is the sum of our own outputs minus the sum of our own inputs?

File: boltzrpc/boltzrpc.proto [hunk 6]
```diff
@@ -733,6 +738,49 @@ message GetWalletRequest {
...
+message TransactionSwap {
+    string id = 1;
+    TransactionType type = 2;
+}
+
+message TransactionOutput {
+    string address = 1;
+    uint64 amount = 2;
+    // wether the address is controlled by the wallet
+    bool is_our_address = 3;
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:777]:
```suggestion
    bool is_ours = 3;
```

File: boltzrpc/boltzrpc.proto [hunk 7]
```diff
@@ -733,6 +738,49 @@ message GetWalletRequest {
...
+    CLAIM = 3;
+}
+
+message WalletTransaction {
+    string id = 1;
+    // balance change of the wallet in satoshis.
+    // its the sum of all output values minus the sum of all input values which are controlled by the wallet.
+    // positive values indicate incoming transactions, negative values outgoing transactions
+    int64 balance_change = 2;
+    int64 timestamp = 3;
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:761]:
```suggestion
    uint64 timestamp = 3;
```

File: onchain/wallet/wallet.go [hunk 8]
```diff
@@ -597,36 +598,17 @@ func (wallet *Wallet) GetBalance() (*onchain.Balance, error) {
...
-		} `json:"transactions"`
-	}
-	if err := withAuthHandler(C.GA_get_transactions(wallet.session, params, handler), handler, &outputs); err != nil {
+	transactoins, err := wallet.GetTransactions(0, 0)
+	if err != nil {
 		return nil, err
 	}
-	for _, tx := range outputs.Transactions {
-		if tx.TxId == txId || txId == "" {
+	for _, tx := range transactoins {
```
Comment (michael1011) [inline onchain/wallet/wallet.go:605]:
```suggestion
	for _, tx := range transactions {
```

File: onchain/wallet/wallet.go [hunk 9]
```diff
@@ -786,6 +768,73 @@ func (wallet *Wallet) SetSpentOutputs(outputs []string) {
...
 
+func (wallet *Wallet) GetTransactions(limit, offset uint64) ([]*onchain.WalletTransaction, error) {
+	if wallet.subaccount == nil {
+		return nil, ErrSubAccountNotSet
+	}
+	if limit == 0 {
+		limit = 30
+	}
+	if limit > 30 {
+		return nil, errors.New("limit cant be larger than 30")
```
Comment (michael1011) [inline onchain/wallet/wallet.go:779]:
```suggestion
		return nil, errors.New("limit cannot be larger than 30")
```

File: boltzrpc/boltzrpc.proto [hunk 10]
```diff
@@ -733,6 +738,49 @@ message GetWalletRequest {
...
+    UNKNOWN = 0;
+    LOCKUP = 1;
+    REFUND = 2;
+    CLAIM = 3;
+}
+
+message WalletTransaction {
+    string id = 1;
+    // balance change of the wallet in satoshis.
+    // its the sum of all output values minus the sum of all input values which are controlled by the wallet.
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:758]:
```suggestion
    // It is the sum of all output values minus the sum of all input values which are controlled by the wallet.
```

File: boltzrpc/boltzrpc.proto [hunk 11]
```diff
@@ -733,6 +738,49 @@ message GetWalletRequest {
...
+enum TransactionType {
+    UNKNOWN = 0;
+    LOCKUP = 1;
+    REFUND = 2;
+    CLAIM = 3;
+}
+
+message WalletTransaction {
+    string id = 1;
+    // balance change of the wallet in satoshis.
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:757]:
```suggestion
    // Balance change of the wallet in satoshis.
```

File: boltzrpc/boltzrpc.proto [hunk 12]
```diff
@@ -733,6 +738,49 @@ message GetWalletRequest {
...
+    // its the sum of all output values minus the sum of all input values which are controlled by the wallet.
+    // positive values indicate incoming transactions, negative values outgoing transactions
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:759]:
```suggestion
    // Positive values indicate incoming transactions, negative values outgoing transactions.
```

File: boltzrpc/boltzrpc.proto [hunk 13]
```diff
@@ -733,6 +738,49 @@ message GetWalletRequest {
...
+    int64 balance_change = 2;
+    int64 timestamp = 3;
+    repeated TransactionOutput outputs = 4;
+    uint32 block_height = 6;
+    // swaps which are related to this transaction
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:764]:
```suggestion
    // Swaps which are related to this transaction.
```

File: boltzrpc/boltzrpc.proto [hunk 14]
```diff
@@ -733,6 +738,49 @@ message GetWalletRequest {
...
+
+message TransactionSwap {
+    string id = 1;
+    TransactionType type = 2;
+}
+
+message TransactionOutput {
+    string address = 1;
+    uint64 amount = 2;
+    // wether the address is controlled by the wallet
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:776]:
```suggestion
    // Whether the address is controlled by the wallet.
```

**Review by michael1011 — APPROVED** [review]
utACK

**Pull request merged**